src/main/receiver.cc

```diff
@@ -1,116 +1,42 @@
-#include "../lib/receiverWithPipe.h"
-#include "../lib/receiverWithMq.h"
-#include <iostream>
-#include <string>
-#include <fcntl.h>
-#include <sys/stat.h>
-#include <sys/types.h>
-#include <unistd.h>
-#include <cstdlib>
-#include <fstream>
-#include <getopt.h>
-#include <cstdio>
-
-enum ipcTypes
-{
-    Pipe,
-    Queue,
-    SharedMemory,
-    OtherSituation
-};
-int dealWithArg(int argc, char **argv, std::string &return_fileName, ipcTypes &return_method);
+#include "src/lib/receiverWithPipe.h"
+#include "src/lib/receiverWithMsgq.h"
+#include "src/lib/receiverWithShmem.h"
+#include "src/lib/argsParser.h"
 
 int main(int argc, char **argv)
 {
-    std::string fileName;
-    ipcTypes method;
-    int resultOfDealWithArg = dealWithArg(argc, argv, fileName, method);
-    if (resultOfDealWithArg == -1)
-    {
-        std::cout << "dealWithArg error" << std::endl;
-        return -1;
-    }
+    ArgsParser argsParser(argc,argv);
+    std::string fileName = argsParser.getFileName();
+    ipcType method = argsParser.getMethod();
```

Comment:
why are you copying the method? you can just pass argsParses.getMethod() inside the switch.